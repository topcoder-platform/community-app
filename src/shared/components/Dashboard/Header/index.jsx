/**
 * Child component of Dashboard/Header renders summary of user achievements,
 * money earned and number of active challenges. Also renders special badges
 * based of acheivement data.
*/
import React from 'react';
import PT from 'prop-types';

import Badge, { MAP, XL_MAP } from './Badge';

import './styles.scss';

export default function Header(props) {
  const {
    achievements,
    showXlBadge,
    xlBadge,
  } = props;
  const badges = achievements.filter(x => MAP[x.description]);
  return (
    <div styleName="container">
      <h1 styleName="title">Dashboard</h1>
      <div styleName="badges">
        {
          badges.map(({ description }) => (
            <Badge
              badge={MAP[description]}
              key={description}
              showXl={show => showXlBadge(show && description)}
              title={description}
              xlBadge={xlBadge === description && XL_MAP[xlBadge]}
            />
          ))
        }
      </div>
    </div>
  );
}

Header.defaultProps = {
  achievements: [],
};

Header.propTypes = {
  achievements: PT.arrayOf(PT.object),
  showXlBadge: PT.func.isRequired,
  xlBadge: PT.string.isRequired,
};
