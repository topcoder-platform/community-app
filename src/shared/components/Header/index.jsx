import _ from 'lodash';
import React, { useState, useEffect } from 'react';
import PT from 'prop-types';
import { config } from 'topcoder-react-utils';
import Logo from 'assets/images/tc-logo.svg';

let TopNavRef;
let LoginNavRef;

try {
  // eslint-disable-next-line global-require
  const { TopNav, LoginNav } = require('navigation-component');
  TopNavRef = TopNav;
  LoginNavRef = LoginNav;
} catch (e) {
  // window is undefined
}

const Header = ({ profile }) => {
  const [activeLevel1Id, setActiveLevel1Id] = useState();
  const [path, setPath] = useState();
  const [openMore, setOpenMore] = useState(true);

  const handleChangeLevel1Id = (menuId) => {
    setActiveLevel1Id(menuId);
  };

  const handleCloseOpenMore = () => {
    setOpenMore(false);
  };

  const handleChangeOpenMore = (changedOpenMore) => {
    setOpenMore(changedOpenMore);
  };

  const handleSwitchMenu = () => {
    setActiveLevel1Id(config.HEADER_MENU[0].id);
  };

  let normalizedProfile = profile && _.clone(profile);
  if (profile) {
    normalizedProfile.photoURL = _.has(profile, 'photoURL') ? `${config.CDN.PUBLIC}/avatar/${
      encodeURIComponent(normalizedProfile.photoURL)}?size=32` : '';
  } else {
    normalizedProfile = null;
  }

  useEffect(() => {
    setPath(window.location.pathname);
  }, []);
  if (TopNavRef) {
    return (
      <div>
        <TopNavRef
          menu={config.HEADER_MENU}
          rightMenu={(
            <LoginNavRef
              loggedIn={!_.isEmpty(profile)}
              notificationButtonState="none"
              notifications={[]}
              accountMenu={config.ACCOUNT_MENU}
              switchText={config.ACCOUNT_MENU_SWITCH_TEXT}
              onSwitch={handleSwitchMenu}
              onMenuOpen={handleCloseOpenMore}
              showNotification={false}
              profile={normalizedProfile}
              authURLs={config.HEADER_AUTH_URLS}
            />
          )}
          logo={<Logo />}
          theme={config.HEADER_MENU_THEME}
          currentLevel1Id={activeLevel1Id}
          onChangeLevel1Id={handleChangeLevel1Id}
          path={path}
          openMore={openMore}
          setOpenMore={handleChangeOpenMore}
          loggedIn={!_.isEmpty(profile)}
          profileHandle={profile ? profile.handle : ''}
        />
      </div>
    );
  }

  return (<div />);
};

Header.defaultProps = {
  profile: null,
};

Header.propTypes = {
  profile: PT.shape({
    photoURL: PT.string,
  }),
};

export default Header;
