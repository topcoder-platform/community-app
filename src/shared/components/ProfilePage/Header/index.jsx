/**
 * Profile Header.  Displays the name, country, potrait and quote for the member.
 */
import React from 'react';
import PT from 'prop-types';
import { noop, get } from 'lodash';
import moment from 'moment';

import { getRatingLevel } from 'utils/tc';
import { config, isomorphy } from 'topcoder-react-utils';

import CopilotIcon from 'assets/images/profile/ico-track-copilot.svg';
import DataScienceIcon from 'assets/images/profile/ico-track-data.svg';
import DesignIcon from 'assets/images/profile/ico-track-design.svg';
import DevelopIcon from 'assets/images/profile/ico-track-develop.svg';

import './styles.scss';

let assets;
if (isomorphy.isClientSide()) {
  assets = require.context('assets/images', false, /svg/);
}

const TRACK_LABELS = {
  COPILOT: 'COPILOT',
  DATA_SCIENCE: 'DATA SCIENTIST',
  DESIGN: 'DESIGNER',
  DEVELOP: 'DEVELOPER',
};

const ProfileHeader = ({
  copilot,
  country,
  info,
  onShowBadges,
  showBadgesButton,
  wins,
}) => {
  const defaultProfilePicture = assets('./ico-user-default.svg');

  const { photoURL = defaultProfilePicture } = info;

  const onProfileImageError = (e) => {
    e.target.src = defaultProfilePicture;
  };

  return (
    <div styleName="container">
      <div>
        <img
          src={photoURL}
          onError={onProfileImageError}
          styleName="profile-circle"
          alt="Member Portait"
        />
      </div>
      <div styleName="info">
        <h1 styleName={`handle level-${getRatingLevel(get(info, 'maxRating.rating', 0))}`}>
          {info.handle}
        </h1>
        <h3 styleName="location-challenges">
          {country}
          {Boolean(wins) && (
            <span>
              {' '}
              |
              {' '}
              {wins}
              {' '}
              Wins
            </span>
          )}
        </h3>
        <h3 styleName="tenure">
          Member Since
          {' '}
          {moment(info.createdAt).format('MMMM, YYYY')}
        </h3>
      </div>
      {
        info.tracks && info.tracks.length > 0
        && (
          <div styleName="tracks-links">
            <div styleName="tracks">
              {
                [...info.tracks, ...(copilot ? ['COPILOT'] : [])].map(track => (
                  <a href={`#${track}`} key={track} styleName="track">
                    {track === 'COPILOT' && <CopilotIcon styleName="track-icon" />}
                    {track === 'DATA_SCIENCE' && <DataScienceIcon styleName="track-icon" />}
                    {track === 'DESIGN' && <DesignIcon styleName="track-icon" />}
                    {track === 'DEVELOP' && <DevelopIcon styleName="track-icon" />}
                    <div styleName="text">
                      {TRACK_LABELS[track]}
                    </div>
                  </a>
                ))
              }
            </div>
          </div>
        )
      }
      {info.description && (
        <p styleName="description">
          {info.description}
        </p>
      )}
      <div styleName="links">
        {
          showBadgesButton ? (
            <a
              onClick={() => onShowBadges()}
              onKeyPress={() => onShowBadges()}
              role="link"
              styleName="link badge-link"
              tabIndex="0"
            >
              Badges
            </a>
          ) : null
        }
        <a href={`${config.URL.FORUMS}/?module=History&userID=${info.userId}`} styleName="link">
          Forum Posts
        </a>
      </div>
    </div>
  );
};

ProfileHeader.defaultProps = {
  copilot: false,
  country: '',
  info: {},
  onShowBadges: noop,
  showBadgesButton: false,
  wins: 0,
};

ProfileHeader.propTypes = {
  copilot: PT.bool,
  country: PT.string,
  info: PT.shape(),
  onShowBadges: PT.func,
  showBadgesButton: PT.bool,
  wins: PT.number,
};

export default ProfileHeader;
