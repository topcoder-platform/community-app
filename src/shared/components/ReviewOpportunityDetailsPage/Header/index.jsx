/**
 * Header which displays review payment, challenge phases, and provides an application button
 */
import _ from 'lodash';
import React from 'react';
import PT from 'prop-types';

import { openPositionsByRole } from 'utils/reviewOpportunities';

import ApplyTime from './ApplyTime';
import PaymentInfo from './PaymentInfo';
import PhaseList from './PhaseList';

import './styles.scss';

/**
 * Header Component
 */
const Header = ({
  details, handle, onApply, phasesExpanded, onPhaseExpand,
}) => (
  <div styleName="container">
    <PaymentInfo
      positions={openPositionsByRole(details)}
    />
    <PhaseList
      isExpanded={phasesExpanded}
      onExpand={onPhaseExpand}
      phases={details.challenge.phases}
    />
    <ApplyTime
      openPositions={details.openPositions}
      onApply={onApply}
      hasApplied={Boolean(_.find(details.applications, app => _.toString(app.handle) === _.toString(handle) && app.status !== 'CANCELLED'))}
      startDate={details.startDate}
      completed={details.openPositions === 0}
    />
  </div>
);

/**
 * Prop Validation
 */
Header.propTypes = {
  details: PT.shape().isRequired,
  handle: PT.string.isRequired,
  onApply: PT.func.isRequired,
  onPhaseExpand: PT.func.isRequired,
  phasesExpanded: PT.bool.isRequired,
};

export default Header;
