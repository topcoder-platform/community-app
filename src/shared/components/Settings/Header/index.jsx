/**
 * Child component of Settings/Header renders summary of number of active challenges.
 * Also renders special badges based on achievements data.
*/
import _ from 'lodash';
import React from 'react';
import PT from 'prop-types';

import Badge, { MAP, XL_MAP } from '../../Dashboard/Header/Badge';

import './styles.scss';

export default function Header(props) {
  const {
    profileState,
    showXlBadge,
    xlBadge,
  } = props;
  const achievements = profileState.achievements || [];
  const badges = achievements.filter(x => MAP[x.description]);
  const hasBadges = badges && badges.length;
  return (
    <div styleName="page-state-header">
      <header>
        <div styleName="page-info">
          <h1>Settings</h1>
        </div>
        <div styleName="info">
          <div styleName="badgeItem">
            {
              _.map(badges, ({ description }) => (
                <Badge
                  badge={MAP[description]}
                  key={description}
                  showXl={show => showXlBadge(show && description)}
                  title={description}
                  xlBadge={xlBadge === description && XL_MAP[xlBadge]}
                />
              ))
            }
          </div>
          <div styleName="separator" className={hasBadges ? '' : 'hidden'} />
          <div styleName="item">
            <div styleName="value">
              <p>{_.isNumber(profileState.activeChallengesCount) ? profileState.activeChallengesCount : ''}</p>
            </div>
            <div styleName="title">
              <p>Active Challenges</p>
            </div>
          </div>
        </div>
      </header>
    </div>
  );
}

Header.defaultProps = {
  xlBadge: null,
};

Header.propTypes = {
  profileState: PT.shape().isRequired,
  showXlBadge: PT.func.isRequired,
  xlBadge: PT.string,
};

