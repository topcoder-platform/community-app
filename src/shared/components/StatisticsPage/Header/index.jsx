/**
 * Badges Popup Component.  Renders a tooltip to show the date a badge was earned.
 */
import React from 'react';
import PT from 'prop-types';
import styles from './styles.scss';

class Header extends React.Component {
    render() {
        const {
          title,
          modalOpen,
          numberOfChallenges,
          activeTitle,
        } = this.props;
        return(
            <div styleName="page-state-header">
                <header>
                    <div styleName="page-info">
                        <h1>{title}</h1>
                        <div styleName="nav-right">
                            <i onClick={()=>{modalOpen()}} styleName="fa fa-th">
                            ...
                        </i>
                        </div>
                    </div>
                    <div styleName="info">
                        <div styleName="item">
                            <div styleName="value">
                                <p>{numberOfChallenges}</p>
                            </div>
                            <div styleName="title">
                                <p>{activeTitle}</p>
                            </div>
                        </div>
                    </div>
                </header>
            </div>
        );
    }
}

Header.defaultProps = {
  pageTitle: "",
  modalOpen: null,
  numberOfChallenges: null,
  activeTitle: "",
};

Header.propTypes = {
  pageTitle: PT.string,
  modalOpen: PT.func,
  numberOfChallenges: PT.number,
  activeTitle: PT.string,
};

export default Header;
