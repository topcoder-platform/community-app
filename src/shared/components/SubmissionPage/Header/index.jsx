/**
 * components.page.challenge-details.submission.header.index Header Component
 *
 * Description:
 *   Header for the submission page.  Displays title and allows navigation
 *   back to challenge details.
 */
import PT from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';
import './styles.scss';

/**
 * Header shown on Submissions Page
 */
const Header = ({
  challengeId,
  challengesUrl,
  title,
}) => (
  <div styleName="header">
    <Link to={`${challengesUrl}/${challengeId}`}>
      <span>
&#x2039;
      </span>
      <p>
Back to challenge
      </p>
    </Link>
    <h1>
      {title}
    </h1>
  </div>
);

/**
 * Prop Validation
 */
Header.propTypes = {
  challengeId: PT.number.isRequired,
  challengesUrl: PT.string.isRequired,
  title: PT.string.isRequired,
};

export default Header;
