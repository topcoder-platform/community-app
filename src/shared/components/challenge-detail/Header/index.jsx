/**
 * Challenge header component.
 * This component renders all other child components part of the header.
 * Any data massaging needed for a child view should be done here.
 */

import _ from 'lodash';
import moment from 'moment';
import 'moment-duration-format';

import PT from 'prop-types';
import React from 'react';
import { DangerButton, PrimaryButton } from 'topcoder-react-ui-kit';
import { Link } from 'topcoder-react-utils';

import LeftArrow from 'assets/images/arrow-prev.svg';

import ArrowUp from '../../../../assets/images/icon-arrow-up.svg';
import ArrowDown from '../../../../assets/images/icon-arrow-down.svg';

import Prizes from './Prizes';
import ChallengeTags from './ChallengeTags';
import DeadlinesPanel from './DeadlinesPanel';
import TabSelector from './TabSelector';

import style from './style.scss';

/* Holds day and hour range in ms. */
const HOUR_MS = 60 * 60 * 1000;
const DAY_MS = 24 * HOUR_MS;

export default function ChallengeHeader(props) {
  const {
    challenge,
    challengesUrl,
    checkpoints,
    hasRegistered,
    numWinners,
    onSelectorClicked,
    onToggleDeadlines,
    registering,
    registerForChallenge,
    setChallengeListingFilter,
    unregisterFromChallenge,
    unregistering,
    challengeSubtracksMap,
    selectedView,
    showDeadlineDetail,
    hasFirstPlacement,
    isMenuOpened,
  } = props;

  const {
    drPoints,
    id: challengeId,
    name,
    subTrack,
    track,
    pointPrizes,
    events,
    technologies,
    platforms,
    prizes,
    numberOfCheckpointsPrizes,
    topCheckPointPrize,
    reliabilityBonus,
    userDetails,
    currentPhases,
    numRegistrants,
    numSubmissions,
    allPhases,
    status,
    appealsEndDate,
  } = challenge;

  const phases = {};
  if (allPhases) {
    allPhases.forEach((phase) => {
      phases[_.camelCase(phase.phaseType)] = phase;
    });
  }

  let registrationEnded = true;
  const regPhase = phases.registration;
  if (status !== 'COMPLETED' && regPhase) {
    registrationEnded = regPhase.phaseStatus !== 'Open';
  }

  const submissionEnded = status === 'COMPLETED'
    || (_.get(phases, 'submission.phaseStatus') !== 'Open'
      && _.get(phases, 'checkpointSubmission.phaseStatus') !== 'Open');

  let trackLower = track ? track.toLowerCase() : 'design';
  if (technologies.includes('Data Science')) {
    trackLower = 'datasci';
  }

  const eventNames = (events || []).map((event => (event.eventName || '').toUpperCase()));

  const miscTags = _.union(
    _.isArray(technologies) ? technologies : (technologies || '').split(', '),
    _.isArray(platforms) ? platforms : (platforms || '').split(', '),
  );

  let bonusType = '';
  if (numberOfCheckpointsPrizes && topCheckPointPrize) {
    bonusType = 'Bonus';
  } else if (reliabilityBonus && reliabilityBonus.toFixed() !== '0') {
    bonusType = 'Reliability Bonus';
  }

  /* userDetails.hasUserSubmittedForReview does not reset to false
   * if the user has deleted all of their submissions, so we have to
   * iterate through all their submissions and ensure that all of them
   * are Deleted
  */
  const hasSubmissions = userDetails && (userDetails.submissions || []).reduce((acc, submission) => acc || submission.status !== 'Deleted', false);

  let nextPhase = (currentPhases && currentPhases[0]) || {};
  if (hasRegistered && nextPhase.phaseType === 'Registration') {
    nextPhase = currentPhases[1] || {};
  }
  const nextDeadline = nextPhase.phaseType;

  const deadlineEnd = moment(nextPhase && nextPhase.scheduledEndTime);
  const currentTime = moment();

  let timeLeft = deadlineEnd.isAfter(currentTime)
    ? deadlineEnd.diff(currentTime) : 0;

  let format;
  if (timeLeft > DAY_MS) format = 'D[d] H[h]';
  else if (timeLeft > HOUR_MS) format = 'H[h] m[min]';
  else format = 'm[min] s[s]';

  timeLeft = moment.duration(timeLeft).format(format);

  let relevantPhases = [];

  if (showDeadlineDetail) {
    relevantPhases = (allPhases || []).filter((phase) => {
      if (phase.phaseType === 'Iterative Review') {
        const end = phase.actualEndTime || phase.scheduledEndTime;
        return moment(end).isAfter(moment());
      }
      const phaseLowerCase = phase.phaseType.toLowerCase();
      if (phaseLowerCase.includes('screening') || phaseLowerCase.includes('specification')) {
        return false;
      }
      if (phaseLowerCase.includes('registration') || phaseLowerCase.includes('checkpoint')
        || phaseLowerCase.includes('submission') || phaseLowerCase.includes('review')) {
        return true;
      }
      return false;
    });

    relevantPhases.sort((a, b) => {
      if (a.phaseType.toLowerCase().includes('registration')) {
        return -1;
      }
      if (b.phaseType.toLowerCase().includes('registration')) {
        return 1;
      }
      return (new Date(a.actualEndTime || a.scheduledEndTime)).getTime()
        - (new Date(b.actualEndTime || b.scheduledEndTime)).getTime();
    });
    if (subTrack === 'FIRST_2_FINISH' && status === 'COMPLETED') {
      const phases2 = allPhases.filter(p => p.phaseType === 'Iterative Review' && p.phaseStatus === 'Closed');
      const endPhaseDate = Math.max(...phases2.map(d => new Date(d.scheduledEndTime)));
      relevantPhases = _.filter(relevantPhases, p => (p.phaseType.toLowerCase().includes('registration')
        || new Date(p.scheduledEndTime).getTime() < endPhaseDate));
      relevantPhases.push({
        id: -1,
        phaseType: 'Winners',
        scheduledEndTime: endPhaseDate,
      });
    } else if (relevantPhases.length > 1 && appealsEndDate) {
      const lastPhase = relevantPhases[relevantPhases.length - 1];
      const lastPhaseTime = (
        new Date(lastPhase.actualEndTime || lastPhase.scheduledEndTime)
      ).getTime();
      const appealsEnd = (new Date(appealsEndDate).getTime());
      if (lastPhaseTime < appealsEnd) {
        relevantPhases.push({
          id: -1,
          phaseType: 'Winners',
          scheduledEndTime: appealsEndDate,
        });
      }
    }
  }

  const checkpointCount = checkpoints && checkpoints.numberOfUniqueSubmitters;

  let nextDeadlineMsg;
  switch ((status || '').toLowerCase()) {
    case 'active':
      nextDeadlineMsg = (
        <div styleName="next-deadline">
          Next Deadline:
          {' '}
          {
            <span styleName="deadline-highlighted">
              {nextDeadline || '-'}
            </span>
          }
        </div>
      );
      break;
    case 'completed':
      nextDeadlineMsg = (
        <div styleName="completed">
          The challenge is finished.
        </div>
      );
      break;
    default:
      nextDeadlineMsg = (
        <div>
          Status:
          &zwnj;
          {
            <span styleName="deadline-highlighted">
              {_.upperFirst(_.lowerCase(status))}
            </span>
          }
        </div>
      );
      break;
  }

  // Legacy MMs have a roundId field, but new MMs do not.
  // This is used to disable registration/submission for legacy MMs.
  const isLegacyMM = subTrack === 'MARATHON_MATCH' && Boolean(challenge.roundId);

  if (hasFirstPlacement && !_.isEmpty(currentPhases)) {
    _.some(currentPhases, { phaseType: 'Final Fix', phaseStatus: 'Open' });
  }

  return (
    <div styleName="challenge-outer-container">
      <div styleName="important-detail">
        <div styleName="title-wrapper" aria-hidden={isMenuOpened}>
          <Link to={challengesUrl}>
            <LeftArrow styleName="left-arrow" />
          </Link>
          <div>
            <h1 styleName="challenge-header">
              {name}
            </h1>
            <ChallengeTags
              subTrack={subTrack}
              track={trackLower}
              challengesUrl={challengesUrl}
              challengeSubtracksMap={challengeSubtracksMap}
              events={eventNames}
              technPlatforms={miscTags}
              setChallengeListingFilter={setChallengeListingFilter}
            />
          </div>
        </div>
        <div styleName="prizes-ops-container">
          <div styleName="prizes-outer-container">
            <h3 styleName="prizes-title">
PRIZES
            </h3>
            <Prizes prizes={prizes && prizes.length ? prizes : [0]} pointPrizes={pointPrizes} />
            {
              bonusType ? (
                <div id={`bonus-${trackLower}`} styleName="bonus-div">
                  {
                    bonusType === 'Bonus'
                      ? (
                        <p styleName="bonus-text">
                          <span styleName={`bonus-highlight ${trackLower}-accent-color`}>
                          BONUS:
                            {' '}
                            {numberOfCheckpointsPrizes}
                          </span>
                        &zwnj;
                        CHECKPOINTS AWARDED WORTH
                        &zwnj;
                          <span
                            styleName={`bonus-highlight ${trackLower}-accent-color`}
                          >
                          $
                            {topCheckPointPrize}
                          </span>
                        &zwnj;
                        EACH
                        </p>
                      )
                      : (
                        <p styleName="bonus-text">
                          <span styleName={`bonus-highlight ${trackLower}-accent-color`}>
                          RELIABILITY BONUS: $
                            {reliabilityBonus.toFixed()}
                          </span>
                        </p>
                      )
                  }
                </div>
              ) : null
            }
            {
              drPoints ? (
                <div styleName="bonus-div">
                  <p styleName="bonus-text">
                    <span styleName={`bonus-highlight ${trackLower}-accent-color`}>
POINTS:
                      {drPoints}
                    </span>
                  </p>
                </div>
              ) : null
            }
          </div>
          <div styleName="challenge-ops-wrapper">
            <div styleName="challenge-ops-container">
              {hasRegistered ? (
                <DangerButton
                  disabled={unregistering || registrationEnded
                  || hasSubmissions || isLegacyMM}
                  forceA
                  onClick={unregisterFromChallenge}
                  theme={{ button: style.challengeAction }}
                >
Unregister
                </DangerButton>
              ) : (
                <PrimaryButton
                  disabled={registering || registrationEnded || isLegacyMM}
                  forceA
                  onClick={registerForChallenge}
                  theme={{ button: style.challengeAction }}
                >
Register
                </PrimaryButton>
              )}
              <PrimaryButton
                disabled={!hasRegistered || unregistering || submissionEnded || isLegacyMM}
                theme={{ button: style.challengeAction }}
                to={`${challengesUrl}/${challengeId}/submit`}
              >
Submit
              </PrimaryButton>
              {
                track === 'DESIGN' && hasRegistered && !unregistering
                && hasSubmissions && (
                  <PrimaryButton
                    theme={{ button: style.challengeAction }}
                    to={`${challengesUrl}/${challengeId}/my-submissions`}
                  >
View Submissions
                  </PrimaryButton>
                )
              }
            </div>
          </div>
        </div>
        <div styleName="deadlines-view">
          <div styleName="deadlines-overview">
            <div styleName="deadlines-overview-text">
              {nextDeadlineMsg}
              {
                (status || '').toLowerCase() === 'active'
                && (
                <div styleName="current-phase">
                  <span styleName="deadline-highlighted">
                    {timeLeft}
                  </span>
                  {' '}
until current deadline ends
                </div>
                )
              }
            </div>
            <a
              onClick={onToggleDeadlines}
              onKeyPress={onToggleDeadlines}
              role="button"
              styleName="deadlines-collapser"
              tabIndex={0}
            >
              {showDeadlineDetail
                ? (
                  <span styleName="collapse-text">
Hide Deadlines
                    <ArrowDown />
                  </span>
                )
                : (
                  <span styleName="collapse-text">
Show Deadlines
                    <ArrowUp />
                  </span>
                )
              }
            </a>
          </div>
          {
            showDeadlineDetail
            && <DeadlinesPanel deadlines={relevantPhases} />
          }
        </div>
        <TabSelector
          challenge={challenge}
          onSelectorClicked={onSelectorClicked}
          trackLower={trackLower}
          selectedView={selectedView}
          numRegistrants={numRegistrants}
          numWinners={numWinners}
          hasCheckpoints={checkpoints && checkpoints.length > 0}
          numSubmissions={numSubmissions}
          hasRegistered={hasRegistered}
          checkpointCount={checkpointCount}
        />
      </div>
    </div>
  );
}

ChallengeHeader.defaultProps = {
  checkpoints: {},
  isMenuOpened: false,
};

ChallengeHeader.propTypes = {
  checkpoints: PT.shape(),
  challenge: PT.shape({
    id: PT.number.isRequired,
  }).isRequired,
  challengesUrl: PT.string.isRequired,
  hasRegistered: PT.bool.isRequired,
  numWinners: PT.number.isRequired,
  onSelectorClicked: PT.func.isRequired,
  onToggleDeadlines: PT.func.isRequired,
  registerForChallenge: PT.func.isRequired,
  registering: PT.bool.isRequired,
  selectedView: PT.string.isRequired,
  setChallengeListingFilter: PT.func.isRequired,
  showDeadlineDetail: PT.bool.isRequired,
  unregisterFromChallenge: PT.func.isRequired,
  unregistering: PT.bool.isRequired,
  challengeSubtracksMap: PT.shape().isRequired,
  hasFirstPlacement: PT.bool.isRequired,
  isMenuOpened: PT.bool,
};
