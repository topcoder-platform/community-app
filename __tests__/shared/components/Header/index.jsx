import React from 'react';
import Renderer from 'react-test-renderer/shallow';

import Header from 'components/Header';

import profileState from './__mocks__/profile-state.json';


const rnd = new Renderer();

it('Default render', () => {
  rnd.render((<Header
    profile={profileState}
  />));
  expect(rnd.getRenderOutput()).toMatchSnapshot();
});
