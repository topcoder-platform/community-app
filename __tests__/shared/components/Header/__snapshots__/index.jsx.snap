// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default render 1`] = `
<div>
  <TopNav
    loggedIn={true}
    logo={
      <Logo
        height="20"
        viewBox="0 0 54 20"
        width="54"
        xmlns="http://www.w3.org/2000/svg"
      />
    }
    menu={
      Array [
        Object {
          "href": "https://www.topcoder-dev.com",
          "id": "business",
          "title": "BUSINESS",
        },
        Object {
          "id": "community",
          "secondaryMenuForGuest": Array [
            Object {
              "href": "https://www.topcoder-dev.com/about",
              "title": "Overview",
            },
            Object {
              "href": "https://www.topcoder-dev.com/how-it-works/faqs/",
              "title": "How It Works",
            },
            Object {
              "href": "/community/learn",
              "title": "Tracks",
            },
            Object {
              "href": "https://www.topcoder-dev.com/about/why-crowdsourcing/",
              "title": "Why Join",
            },
          ],
          "secondaryMenuForLoggedInUser": Array [
            Object {
              "href": "/my-dashboard",
              "title": "Dashboard",
            },
            Object {
              "href": "/members/willFilledByUserName",
              "id": "myprofile",
              "title": "My Profile",
            },
            Object {
              "href": "https://community.topcoder-dev.com/PactsMemberServlet?module=PaymentHistory&full_list=false",
              "title": "Payments",
            },
          ],
          "subMenu": Array [
            Object {
              "subMenu": Array [
                Object {
                  "href": "/challenges",
                  "title": "All Challenges",
                },
                Object {
                  "href": "https://arena.topcoder-dev.com",
                  "title": "Competitive Programming",
                },
              ],
              "title": "Compete",
            },
            Object {
              "subMenu": Array [
                Object {
                  "href": "/community/competitive-programming",
                  "title": "Competitive Programming",
                },
                Object {
                  "href": "/community/data-science",
                  "title": "Data Science",
                },
                Object {
                  "href": "/community/design",
                  "title": "Design",
                },
                Object {
                  "href": "/community/development",
                  "title": "Development",
                },
                Object {
                  "href": "/community/qa",
                  "title": "QA",
                },
              ],
              "title": "Tracks",
            },
            Object {
              "subMenu": Array [
                Object {
                  "href": "https://www.topcoder-dev.com/tco",
                  "title": "TCO",
                },
                Object {
                  "href": "/community/member-programs",
                  "title": "Programs",
                },
                Object {
                  "href": "https://apps.topcoder-dev.com/forums",
                  "title": "Forums",
                },
                Object {
                  "href": "/community/statistics",
                  "title": "Statistics",
                },
                Object {
                  "href": "/community/events",
                  "title": "Events",
                },
                Object {
                  "href": "https://www.topcoder-dev.com/blog",
                  "title": "Blog",
                },
              ],
              "title": "Explore",
            },
          ],
          "title": "COMMUNITY",
        },
      ]
    }
    onChangeLevel1Id={[Function]}
    openMore={true}
    profileHandle="huanner"
    rightMenu={
      <LoginNav
        accountMenu={
          Array [
            Object {
              "href": "/settings/profile",
              "title": "Settings",
            },
            Object {
              "separator": true,
            },
            Object {
              "href": "https://help.topcoder-dev.com/",
              "title": "Help",
            },
            Object {
              "href": "https://www.topcoder.com/about/",
              "title": "About Topcoder",
            },
            Object {
              "href": "https://www.topcoder-dev.com/logout",
              "title": "Log Out",
            },
          ]
        }
        authURLs={
          Object {
            "href": "https://accounts.topcoder-dev.com/member/registration?utm_source=community-app-main",
            "location": "https://accounts.topcoder-dev.com/member?retUrl=%S&utm_source=community-app-main",
          }
        }
        loggedIn={true}
        notificationButtonState="none"
        notifications={Array []}
        onMenuOpen={[Function]}
        onSwitch={[Function]}
        profile={
          Object {
            "description": "devxxxxxx",
            "email": "liwanxiang@gmail.com",
            "handle": "huanner",
            "photoURL": "https://d1aahxkjiobka8.cloudfront.net/avatar/https%3A%2F%2Ftopcoder-dev-media.s3.amazonaws.com%2Fmember%2Fprofile%2Fhuanner-1552562543506.png?size=32",
          }
        }
        showNotification={false}
        switchText={
          Object {
            "href": "https://connect.topcoder-dev.com",
            "title": "Switch to BUSINESS",
          }
        }
      />
    }
    setOpenMore={[Function]}
    theme="light"
  />
</div>
`;
